src/backend/backend.c

```diff
@@ -153,7 +153,7 @@ void paint_all_new(session_t *ps, struct managed_win *t) {
 		//
 		// TODO(yshui): maybe we don't need to resize reg_damage, only reg_paint?
 		int resize_factor = 1;
-		if (t) {
+		if (t && t->stacking_rank > 1) {
 			resize_factor = t->stacking_rank;
```

Comment:
hmm, maybe this should be `t->stacking_rank + 1`?

Reply:
if you mean `resize_factor = t->stacking_rank + 1` then i think that's too pessimistic and we're already pessimistic according to the comment above.

Reply:
makes sense.